linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1642,7 +1642,7 @@ private[optimizer] abstract class OptimizerCore(
         if (used.value.isUsed) {
           val ident = LocalIdent(name)
           resolveLocalDef(value) match {
-            case PreTransRecordTree(valueTree, valueTpe, cancelFun) =>
+            case PreTransRecordTree(valueTree, valueStructure, cancelFun) =>
               val recordType = valueTree.tpe.asInstanceOf[RecordType]
               if (!isImmutableType(recordType))
```

Comment:
- Take the `recordType` from the structure so you don't have to cast?
- Assert `valueTree.tpe == valueStructure.recordType`?
- if not, it seems `valueStructure` is unused.